### PR TITLE
Add BSC integration

### PR DESCRIPTION
**Pull request by leetdev**
Add support for BSC as per #335:
- Deployed contracts on both BSC Mainnet and Testnet and added addresses to README
- Added BSC HMT token contract (deployed by Multichain)
- Added subgraph deployment to GitHub graph deploy workflow (please note that `humanprotocol/bsc` subgraph needs to be created on thegraph hosted service)
- Added link to my GitHub profile in the network maintainers list

Additionally, improved the truffle `deploy_contracts` migration script to include deployment of the EscrowFactory contract.

### CODE REVIEW

File: migrations/3_deploy_escrow_factory.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const EscrowFactory = artifacts.require('./EscrowFactory.sol');
+
+const ADDRESS_INPUT_FILENAME = './deployed-hmtoken/hmt.address.json';
+const ADDRESS_OUTPUT_FILENAME = './deployed-escrow-factory/escrow-factory.address.json';
+
+module.exports = (deployer) => {
+  const { address } = JSON.parse(fs.readFileSync(ADDRESS_INPUT_FILENAME).toString());
+
+  deployer
```
Comment (foufrix) [inline migrations/3_deploy_escrow_factory.js]: all of this can go directly in 2_deploy_contracts.js after HMT deployment

**Comment by Dzeranov** [comment]
@leetdev 
1. Please add a proper description to this PR.
2. Add BSC contracts source code (like it was previously done for polygon contracts).

**Comment by leetdev** [comment]
@Dzeranov,

1. I've updated the PR description.
2. Do you mean to the network explorer? If so, then I've already done that, see:
  * EscrowFactory: [0x161228DaAB6F935548E2B98f6cedC9FF648C781d](https://bscscan.com/address/0xc88bC422cAAb2ac8812de03176402dbcA09533f4#code)
  * KVStore: [0x8340412Ed68BcF53a7Da72BFFc1E2E74CfdE74D0](https://bscscan.com/address/0x8340412Ed68BcF53a7Da72BFFc1E2E74CfdE74D0#code)

**Comment by Dzeranov** [comment]
@leetdev ,

No, I mean to this repo. As you can see we have a folder `polygon_contracts` in this repo. So we need a similar one for bsc.

**Comment by leetdev** [comment]
@Dzeranov I'm not sure what needs to go into that folder. Polygon I believe has its own underlying token contract for the Multichain bridge, but I saw no need to create one for BSC. The only contracts I deployed there were the Escrow Factory and the KV Store.

**Comment by Dzeranov** [comment]
@leetdev 
If we have a contract (in any chain) which source code is different from the one you can see in `contracts` folder, it should be represented in this repo for at least two reasons:
1. We need to review it.
2. It's inconvenient to check the source code in the explorer.

**Comment by leetdev** [comment]
@Dzeranov The HMT contract on BSC was deployed by Multichain as part of the bridge setup. Should I look for the source code for that and add it here?

**Comment by Dzeranov** [comment]
@leetdev 
Yes please.

**Comment by leetdev** [comment]
@Dzeranov I've added the [token contract source code](https://github.com/anyswap/anyswap-v1-core/blob/master/contracts/AnyswapV6ERC20.sol) that was deployed by Multichain (verifiable at https://bscscan.com/address/0x0d501B743F22b641B8C8dfe00F1AAb881D57DDC7#code).

**Review by foufrix — APPROVED** [review]

**Review by foufrix — APPROVED** [review]

**Comment by 0xVoronov** [comment]
Hi @leetdev!
Thank you for your work! 
We noticed that the contract is not upgradable, could you also do a proxy implementation, as is done for the [polygon_contracts](https://github.com/humanprotocol/hmt-escrow/tree/master/polygon_contracts)?

**Comment by leetdev** [comment]
@eugenvoronov, I'm not sure what exactly you want me to do. I explained, that the polygon bridge was set up in a different way - Multichain supports both bridging to an existing token (as is the case with Polygon) as well as creating a bridge without an underlying token contract. I was not in charge of setting up the bridge, all I did was suggest using Multichain. There are other HMT bridges that have been set up using the latter model, and I don't see any proxy contracts for those. Am I missing something?

**Comment by posix4e** [comment]
@leetdev i need to discuss this change with you can you reach out over discord?

**Pull request merged**